pkg/utils/kube/kube.go

```diff
@@ -172,14 +172,19 @@ func IsCRD(obj *unstructured.Unstructured) bool {
 	return IsCRDGroupVersionKind(obj.GroupVersionKind())
 }
 
+// ServerResourceForGroupVersionKind looks up and returns the API resource from
+// the server for a given GVK scheme. If verb is set to the non-empty string,
+// it will return the API resource which supports the verb. There are some edge
+// cases, where the same GVK is represented by more than one API.
+//
 // See: https://github.com/ksonnet/ksonnet/blob/master/utils/client.go
-func ServerResourceForGroupVersionKind(disco discovery.DiscoveryInterface, gvk schema.GroupVersionKind) (*metav1.APIResource, error) {
+func ServerResourceForGroupVersionKind(disco discovery.DiscoveryInterface, gvk schema.GroupVersionKind, verb string) (*metav1.APIResource, error) {
 	resources, err := disco.ServerResourcesForGroupVersion(gvk.GroupVersion().String())
 	if err != nil {
 		return nil, err
 	}
 	for _, r := range resources.APIResources {
-		if r.Kind == gvk.Kind {
+		if r.Kind == gvk.Kind && isSupportedVerb(&r, verb) {
```

Comment:
I'm not sure if users will ever run into a misleading message where we report `NotFound` error in logs but it's not because the GVK doesn't exist, but rather the verb isn't supported and we did not distinguish between the two. Since it's a pretty rare corner case, I don't think it's necessary but wanted to point it out.

Reply:
I actually agree with this comment, @jessesuen. Edge case or not, I'm going to fix this.

Reply:
Changed the logic to return `MethodNotSupported` if the GVK resolves to a resource which does not support the given verb.

Also, added some unit tests around all that.